[bug/qa] Fix lint errors being ignored #485

ssmadhavan006 opened the pull request:
## Checklist

- [x] I have read the [OpenWISP Contributing Guidelines](http://openwisp.io/docs/developer/contributing.html).
- [x] I have manually tested the changes proposed in this pull request.
- [x] I have written new test cases for new code and/or updated existing tests for changes to existing code.
- [ ] I have updated the documentation.

## Reference to Existing Issue

Closes #485.

## Description of Changes
commit - fb328699a0a6fc8081e70b08e90b8eb88058fe0f
- Removed broken lint blocks from `molecule/default/molecule.yml` and `molecule/local/molecule.yml`
- Edited `run-qa-checks` script by adding yamllint and ansible-lint
- Add `.yamllint.yml` config to exclude virtual environments and and irrelevant files
- Update CI 'ci.yml' with `continue-on-error:` true and added Added conditional test execution based on dependency installation success
- Fixed indentation in `molecule/resources/converge.yml`
- Removed trailing spaces in `vars/debian-13.yml`
- Converted all YAML files from CRLF to LF line endings
- Lint errors are no longer ignored, failures properly reported


commit 986f4669a68e03a2579161e172e9e489f31e75f5
- Updated Molecule callback configuration to replace deprecated `yaml` stdout_callback with
`ansible.builtin.default` and `result_format: yaml`, resolving CI failures due to removal of
community.general.yaml removal in v12+.

## Screenshot

N/A

ssmadhavan006 commented:
Hi @nemesifier ,

Just a quick update from my side:

After my original PR (fb32869) failed due to the Molecule deprecation issue (which was unrelated to the lint fixes), I added a follow-up commit (986f466) that updates the Molecule callback configuration and resolves the CI failure. You kindly triggered the GitHub Actions run for me, and all tests passed successfully. However, a recent commit in the repository has now introduced a merge conflict. Whenever you get a moment, could you please take a look at the PR again? All CI checks were previously green with my fix, so I can rebase or update the PR as needed once you confirm. No rush  just wanted to keep everything on your radar. 
Thanks you for your time!!!

pandafy commented on molecule/local/molecule.yml:
```diff
@@ -6,10 +6,6 @@ dependency:
-lint: |
-  set -e
-  yamllint . || true
-  ansible-lint || true
```
Question: Why do we remove these?

pandafy reviewed (commented):
Thanks for the PR @ssmadhavan006! I have some suggestions below.  

Rebasing on the current master should also resolve the conflicts.

nemesifier reviewed (commented):
There's some conflicts too that need to be resolved, but let's focus on fixing the QA checks and lint errors cleanly first.

ssmadhavan006 commented:
Hi @nemesifier ,

I’ve pushed an updated set of changes addressing the points you highlighted.

Specifically:

- Removed the deprecated Molecule lint: blocks that were both non-functional and suppressing failures.
- Moved all linting to run-qa-checks and ensured failures are no longer ignored.
- Updated CI so QA checks are enforced and no longer pass silently on errors.
- Resolved the merge conflicts and aligned the branch with the latest changes you pushed.

I also ran the QA checks locally to verify that linting is actually executed now, not just added to the pipeline. The remaining RST warning comes from the existing documentation formatter check (docstrfmt) and does not affect the lint enforcement logic introduced by this PR; I’ve kept the scope focused on fixing ignored lint errors as requested.

Thanks for your patience and for pushing the helper commits, they clarified the expected direction. Please let me know if you’d like me to handle the docstrfmt cleanup in a separate PR or adjust anything further here.

<img width="984" height="251" alt="image" src="https://github.com/user-attachments/assets/f39857fd-5caa-426c-9da1-765dc7f91ff4" />

Varadraj75 commented:
Hi @nemesifier, I’m picking this up and continuing the work on #485.
I’ve reviewed the previous discussion and focused on ensuring QA checks are enforced correctly without expanding scope.
I’ll push incremental fixes to this PR shortly.
